pkg/dyninst/decode/decoder.go

```diff
@@ -60,478 +73,119 @@ type typeAndAddr struct {
-	case reflect.Uint16:
-		if len(data) != 2 {
-			return errors.New("passed data not long enough for uint16")
-		}
-		return enc.WriteToken(jsontext.Int(int64(uint16(binary.NativeEndian.Uint16(data)))))
-	case reflect.Uint32:
-		if len(data) != 4 {
-			return errors.New("passed data not long enough for uint32")
-		}
-		return enc.WriteToken(jsontext.Int(int64(uint32(binary.NativeEndian.Uint32(data)))))
-	case reflect.Uint64:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for uint64")
-		}
-		return enc.WriteToken(jsontext.Int(int64(binary.NativeEndian.Uint64(data))))
-	case reflect.Uintptr:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for uintptr")
-		}
-		return enc.WriteToken(jsontext.Int(int64(binary.NativeEndian.Uint64(data))))
-	case reflect.Float32:
-		if len(data) != 4 {
-			return errors.New("passed data not long enough for float32")
-		}
-		return enc.WriteToken(jsontext.Float(float64(math.Float32frombits(binary.NativeEndian.Uint32(data)))))
-	case reflect.Float64:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for float64")
-		}
-		return enc.WriteToken(jsontext.Float(math.Float64frombits(binary.NativeEndian.Uint64(data))))
-	case reflect.Complex64:
-		if len(data) != 8 {
-			return errors.New("passed data not long enough for complex64")
-		}
-		realBits := math.Float32frombits(binary.NativeEndian.Uint32(data[0:4]))
-		imagBits := math.Float32frombits(binary.NativeEndian.Uint32(data[4:8]))
-		return enc.WriteToken(jsontext.String(strconv.FormatComplex(complex(float64(realBits), float64(imagBits)), 'f', -1, 64)))
-	case reflect.Complex128:
-		if len(data) != 16 {
-			return errors.New("passed data not long enough for complex128")
-		}
-		realBits := math.Float64frombits(binary.NativeEndian.Uint64(data[0:8]))
-		imagBits := math.Float64frombits(binary.NativeEndian.Uint64(data[8:16]))
-		err := enc.WriteToken(jsontext.Float(float64(realBits)))
-		if err != nil {
-			return err
-		}
-		err = enc.WriteToken(jsontext.Float(float64(imagBits)))
-		if err != nil {
-			return err
-		}
-		return nil
-	case reflect.Array:
-		return errors.New("arrays are not base types")
-	case reflect.Chan:
-		return errors.New("channels are not base types")
-	case reflect.Func:
-		return errors.New("funcs are not base types")
-	case reflect.Interface:
-		return errors.New("interfaces are not base types")
-	case reflect.Map:
-		return errors.New("maps are not base types")
-	case reflect.Ptr:
-		return errors.New("ptrs are not base types")
-	case reflect.Slice:
-		return errors.New("slices are not base types")
-	case reflect.String:
-		return errors.New("strings are not base types")
-	case reflect.Struct:
-		return errors.New("structs are not base types")
-	case reflect.UnsafePointer:
-		return errors.New("unsafe pointers are not base types")
-	default:
-		return errors.New("invalid base type")
-	}
+func (s *snapshotMessage) clear() {
+	s.Debugger.Snapshot = snapshotData{}
```

Comment:
If we're going to preserve and reuse some map data structures, we'll need to do something fancier here.

Reply:
Is there something in particular you're thinking when you say fancy? Or explicitly zeroing/clearing all the fields under s.Debugger.Snapshot is what you were thinking?